st2common/st2common/services/datastore.py

```diff
@@ -211,9 +214,9 @@ def _get_api_client(self):
 
         if not self._client or token_expire:
             self._logger.audit('Creating new Client object.')
-            ttl = (24 * 60 * 60)
+            ttl = cfg.CONF.auth.service_token_ttl
             self._token_expire = get_datetime_utc_now() + timedelta(seconds=ttl)
-            temporary_token = create_token(username=self._api_username, ttl=ttl)
+            temporary_token = create_token(username=self._api_username, ttl=ttl, service=True)
```

Comment:
Passing additional metadata aka the name of the service (action, sensor) where this token is used would be great, but this would require small refactoring - adding additional argument to `DatastoreService` constructor and updating places where it's used (should only be two places - sensor wrapper and python action wrapper).